pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularAOMAddParticipantUpdates.java

```diff
@@ -0,0 +1,513 @@
+
+        return Optional.ofNullable(userSql.findByUserGuid(userGuid));
+    }
+
+    private String getTargetStudyGuid() {
+        return this.patchConfig.getString(Metadata.STUDY.key);
+    }
+
+    private Optional<FormSectionDef> sectionForBlock(FormActivityDef activityDefinition, String blockGuid) {
+        return activityDefinition.getAllSections().stream()
+            // A block guid should belong to one, and only one, section
+            .filter((section) -> section.getBlocks().stream().anyMatch((block) -> block.getBlockGuid().equals(blockGuid)))
+            .findFirst();
+    }
+
+    private List<QuestionBlockDef> findBlocksForQuestions(FormActivityDef activityDefinition, Collection<String> stableIds) {
+        final var stableIdSet = new HashSet<>(stableIds);
+
+        return activityDefinition.getAllSections().stream()
+            .map((section) -> section.getBlocks())
+            .flatMap(List::stream)
+            .map((block) -> {
+                // Relies on container blocks not being themselves nested.
+                // If that assumption changes, this code may fail to find
+                // specific questions if they are nested too deep.
+                switch (block.getBlockType()) {
+                    case QUESTION:
+                        return (List<FormBlockDef>)(List.of(block));
+                    case CONDITIONAL:
+                        var conditionalBlock = ConditionalBlockDef.class.cast(block);
+                        return (List<FormBlockDef>)conditionalBlock.getNested();
+                    case GROUP:
+                        var groupBlock = GroupBlockDef.class.cast(block);
+                        return (List<FormBlockDef>)groupBlock.getNested();
+                    case TABULAR:
+                        var tabularBlock = TabularBlockDef.class.cast(block);
+                        return tabularBlock.getBlocks();
+                    default:
+                        return new ArrayList<FormBlockDef>();
+                }
+            })
+            .flatMap(List::stream)
+            .filter((block) -> block.getBlockType() == BlockType.QUESTION)
+            .map((block) -> QuestionBlockDef.class.cast(block))
+            .filter((questionBlock) -> stableIdSet.contains(questionBlock.getQuestion().getStableId()))
+            .collect(Collectors.toList());
+    }
+
+    private FormBlockDef createBlockDef(Config blockConfig) {
+        final var blockJson = ConfigUtil.toJson(blockConfig);
+        final var blockDef = gson.fromJson(blockJson, FormBlockDef.class);
+        final var validationErrors = gsonValidator.validateAsJson(blockDef);
+
+        if (validationErrors.isEmpty()) {
+            // No errors found!
+            // Just return the object
+            return blockDef;
+        }
+
+        final var message = validationErrors.stream()
+                .map(JsonValidationError::toDisplayMessage)
+                .collect(Collectors.joining(", "));
```

Comment:
Using StreamEx would allow to simplify it by `.join(", ")`

Reply:
Yes, but that adds unnecessary complexity by using StreamEx.